src/App.js

```diff
@@ -29,34 +38,51 @@ function App() {
           return response.json();
     })
       .then((json) => {
+        console.log(json);
         setRepos(json);
       })
     .catch((error) => {
-      setError(error.message)
-    })}
+      setError(error.message);
+    })
+  }
       
+    const changeUrl = () => {
+      const params = new URLSearchParams(window.location.search);
+      params.set('query', inputValue);
+      window.history.replaceState(null, '', `?${params.toString()}`);
+      getRepoByUsername();
+    }
+    
+    useEffect(() => {
+        getRepoByUsername(inputValue);
+    }, [inputValue]);
   
   return (
-    <div>
-      <h1>GitHub Repository Explorer</h1>
-      <div>
-        <input type="text" value={inputValue}
-          onChange={(e) => setInputValue(e.target.value)}
-          placeholder="Put username here"/>
-        <button onClick={getRepoByUsername}>Show repos</button>
-      </div>
-      <div >
-      {errorMessage ? (
-        <p>{errorMessage}</p>
-        ) :       repos.map((repo) => (
-          <div className='repos' key={repo.id}>
-            <h3>{repo.name}</h3>
-            <p>{repo.description}</p>
-          </div>
-      ))}
-  </div>
-  </div>
+    <>
+    <Router>
+      <Routes>
+        <Route path='/' 
+        element={
+        <><Input
+          inputValue={inputValue}
+          setInputValue={setInputValue}
+          changeUrl={changeUrl}
+        />
+        <RepoCard 
+          errorMessage={errorMessage}
+          repos={repos}
+        />
+        </>}
+        />
+        <Route path="/commits/:username/:repoName"
+        element={
+          <Commits
+          />}
+        />
+        </Routes>
+      </Router>
+  </>
```

Comment:
```suggestion
    <Router>
      <Routes>
        <Route path='/' 
        element={
       <RepoLookup />}
        />
        <Route path="/commits/:username/:repoName"
        element={
          <Commits
          />}
        />
        </Routes>
      </Router>
  </>
```

and then your actual page goes into RepoLookup component. 
Those routing files should just have the routes